hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs

```diff
@@ -124,28 +132,34 @@ withCachedModuleAndData uri noCache callback = do
              Nothing -> do
                val <- cacheDataProducer cm
                let dat' = Map.insert (typeOf val) (toDyn val) dat
-               modifyCache (\s -> s {uriCaches = Map.insert uri' (UriCache cm dat')
+               modifyCache (\s -> s {uriCaches = Map.insert uri' (UriCache cm dat' False)
                                                                  (uriCaches s)})
                return val
-             Just x -> do
+             Just x ->
                case fromDynamic x of
                  Just val -> return val
                  Nothing  -> error "impossible"
       callback cm a
```

Comment:
Shouldn't this mirror the changes in withCachedModule?

Reply:
Updated this to remove noCache and used `IdeResponseDeferred` instead